src/Core/Algorithms/Legacy/Fields/RegisterWithCorrespondences.cc

```diff
@@ -847,8 +887,50 @@ bool RegisterWithCorrespondencesAlgo::runP(FieldHandle input, FieldHandle Cors1,
       Tmat(2, 1)*traceA*norm1,
       Tmat(2, 2)*traceA*norm1,
       0.0 };
-    
-    //done with solve, make the new field
+  
+  DenseMatrix rotation(Eigen::MatrixXd::Identity(4,4));
+  DenseMatrix trans1(Eigen::MatrixXd::Identity(4,4));
+  DenseMatrix trans2(Eigen::MatrixXd::Identity(4,4));
+  DenseMatrix scale1(Eigen::MatrixXd::Identity(4,4));
+  DenseMatrix scale2(Eigen::MatrixXd::Identity(4,4));
+  double *r_data = rotation.data();
+  
+  for (int p = 0; p < coefs.size(); p++)
+  {
+    r_data[p] = coefs[p];
+  }
+  // recreate previous moves
+  trans1(0,3) = -sumx2;
+  trans1(1,3) = -sumy2;
+  trans1(2,3) = -sumz2;
+  
+  trans2(0,3) = sumx;
+  trans2(1,3) = sumy;
+  trans2(2,3) = sumz;
+  
+  scale1(0,0) = 1/norm2;
+  scale1(1,1) = 1/norm2;
+  scale1(2,2) = 1/norm2;
+  
+  scale2(0,0) = norm1;
+  scale2(1,1) = norm1;
+  scale2(2,2) = norm1;
+  
+  
+  printMatrix(trans1,"trans1");
+  printMatrix(trans2,"trans2");
+  printMatrix(scale1,"scale1");
+  printMatrix(scale2,"scale2");
+  
+  
+  DenseMatrix transform;
+  transform = trans2*rotation*scale1*trans1;
+  DenseMatrixHandle trans_out(new DenseMatrix(transform.matrix()));
```

Comment:
Should be able to create and do the math on the same line without copying:
`auto trans_out = boost::make_shared<DenseMatrix>(trans2*rotation*scale1*trans1);`